crates/driver/src/driver.rs

```diff
@@ -52,23 +54,25 @@ impl Driver {
             .await?
             .pop()
             .context("simulation returned no results")?;
-        anyhow::ensure!(
-            simulation_details.gas_estimate.is_ok(),
-            "settlement reverted during simulation"
-        );
+        match simulation_details.gas_estimate {
```

Comment:
Seems a bit weird to define the success of simulation based on the `gas_estimate`, in a function where it is not explicitly known what simulation api is used (web3, tenderly, something else...). Not 100% sure, but it seems to me that tenderly api returns `gas_used` for both success and failure. 
Not really an issue, just a nit comment that I would prefer to have some explicit success/failure field instead of using `gas_estimate` maybe...

Reply:
My alternative idea was sth like:
```rust
enum SimulationDetails {
    Success {
        settlement: Settlement,
        solver: Arc<dyn Solver>,
        access_list: Option<AccessList>,
        gas_estimate: U256
    },
    Failure {
        settlement: Settlement,
        solver: Arc<dyn Solver>,
        access_list: Option<AccessList>,
        error: ExecutionError
    }
}
```
but that duplicated the members and ended up in huge pattern matching statements. So I went for the admittedly less idiomatic way.

I could also try to see how the code looks with:
```rust
enum Outcome {
    Success{ gas_estimate: U256 },
    Revert{ error: ExecutionError }
}

struct SimulationDetails {
        settlement: Settlement,
        solver: Arc<dyn Solver>,
        access_list: Option<AccessList>,
        outcome: Outcome
}
```

Or the lowest hanging fruit would be to just add more doc comments on `simulate_settlements()` and `SimulationDetails`.

Reply:
I went with simply adding more detailed docs about the behavior of the functions and meaning of the fields.